[ios][media-library] Include location in getAssetsAsync batch results (#48630)
expo/expo: docs/pages/versions/unversioned/sdk/media-library-legacy.mdx

On Android, when using `getAssetsAsync` without `resolveWithFullInfo: true`, image orientation may be incorrect because EXIF data (which includes orientation) is only read when that option is enabled.

### Performance impact of `resolveWithFullInfo`

> Enabling `resolveWithFullInfo: true` in `getAssetsAsync` significantly increases request time on Android (~5×), because EXIF and location data must be fetched per photo. On iOS, the impact is much smaller (~10–20%), because the data is already included in the query results.

Reviewer: ✏️ **style** `prose.md#voice-and-tone`

"EXIF and location data must be fetched per photo" and "the data is already included in the query results" are passive. Prefer the active forms, and mirror the fix in the two TSDoc notes in **packages/expo-media-library/src/legacy/MediaLibrary.ts** so the API section stays in sync.

```suggestion
Enabling `resolveWithFullInfo: true` in `getAssetsAsync` significantly increases request time on Android (~5×), because the library fetches EXIF and location data per photo. On iOS, the impact is much smaller (~10–20%), because the query results already include the data.
```

<!-- docs-review -->
